### ISSUE TITLE
Report all missing dependencies at once

### ISSUE THREAD

**Issue by DannyBen** [DannyBen/bashly#582]
### Discussed in https://github.com/DannyBen/bashly/discussions/580

<div type='discussions-op-text'>

<sup>Originally posted by **meleu** December 24, 2024</sup>
First of all: **WOW**! :open_mouth:

As bash-lover for more than 2 decades, I'm amazed by this project! This is a life-changing tool! Thanks for sharing it with us! :heart:

Now the feature idea:

---

When the script has more than one dependency and the user doesn't have any of them installed, it is a frustrating experience to see only one "missing dependency" notification at each time you try to run the script.

It would be nice if the dependency checking could report all missing dependencies in a single run.

I'm sharing here what I usually do (in a hope it can give you some ideas):

```bash
# declare the dependencies in an array
readonly DEPENDENCIES=(dep1 dep2 depN)

# checkDependencies:
# prints the missing dependency
# finishes with failure status when a missing dependency is found
checkDependencies() {
  local dep
  local failure=false

  for dep in "${DEPENDENCIES[@]}"; do
    command -v "$dep" > /dev/null && continue
    echo "[ERROR]: '$dep' is not in your PATH" >&2
    failure=true
  done

  [[ "$failure" == 'false' ]]
}
```</div>

**Comment by DannyBen** [comment]
@meleu - this feature is implemented in #583.
Do you wish to test it?

**Comment by meleu** [comment]
sure! I'm on it right now :rocket:
